js_tests/admin/URLify.test.js

```diff
@@ -0,0 +1,29 @@
+/* global QUnit, URLify */
+/* eslint global-strict: 0, strict: 0 */
+'use strict';
+
+QUnit.module('admin.URLify');
+
+QUnit.test('empty string', function(assert) {
+    assert.strictEqual(URLify('', 8, true), '');
+});
+
+QUnit.test('strip non-essential words', function(assert) {
+    // Non-essential words should be stripped prior to truncation.
+    assert.strictEqual(URLify('the D is silent', 8, true), 'd-silent');
+});
+
+QUnit.test('strip non-URL-path characters', function(assert) {
+    // Non-URL characters should be stripped prior to truncation.
+    assert.strictEqual(URLify('D#silent@', 7, true), 'dsilent');
+});
+
+QUnit.test('merge adjacent whitespace', function(assert) {
+    // Adjacent whitespace should be merged prior to truncation.
+    assert.strictEqual(URLify('D   silent', 8, true), 'd-silent');
+});
+
+QUnit.test('no post-truncation hyphen', function(assert) {
+    // Ensure that string truncation does not leave a superfluous trailing hyphen.
```

Comment:
As suggested in [Coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style), state the expected behavior, e.g. "Trailing hyphen are removed." and avoid use of "Ensure that" since all tests ensure stuff.